diffpy/snmf/new_subroutines.py

```diff
@@ -6,8 +6,22 @@
 
 
 def create_components(number_of_components, grid_vector, number_of_signals, signal_length):
+    """Creates the ComponentSignal objects
+
+    Parameters
+    ----------
+    number_of_components: int
+    grid_vector: 1d array
+    number_of_signals: int
+    signal_length: int
+
+    Returns
+    -------
+    tuple of ComponentSignal objects
+
+    """
     component_list = []
-    for c in number_of_components:
+    for c in range(number_of_components):
```

Comment:
I wonder whether it is better to give the components real uuids?  How is that info used later?